kinoml/features/complexes.py

```diff
@@ -0,0 +1,544 @@
+        protein_path = "/path/to/protein.pdb"  # mmcif writing not supported by openeye
+        self._write_molecules([prepared_protein], protein_path)
+        file_protein = FileProtein(path=protein_path)
+
+        ligand_path = "/path/to/ligand.sdf"
+        self._write_molecules(docking_poses, ligand_path)
+        file_ligand = FileLigand(path=ligand_path)
+        protein_ligand_complex = ProteinLigandComplex(
+            components=[file_protein, file_ligand]
+        )
+
+        return protein_ligand_complex
+
+    @staticmethod
+    def _read_ligands(path: str) -> List[oechem.OEGraphMol]:
+        """
+        Read ligands from a file.
+        Parameters
+        ----------
+        path: str
+            Path to ligand file.
+        Returns
+        -------
+        molecules: list of oechem.OEGraphMol
+            A List of ligands as OpenEye molecules.
+        """
+        from openeye import oechem
+
+        molecules = []
+        with oechem.oemolistream(path) as ifs:
+            for molecule in ifs.GetOEGraphMols():
+                molecules.append(oechem.OEGraphMol(molecule))
+
+        # TODO: returns empty list if something goes wrong
+        return molecules
+
+    @staticmethod
+    def _read_protein(path: str) -> oechem.OEGraphMol:
+        """
+        Read a protein from a file.
+        Parameters
+        ----------
+        path: str
+            Path to protein file.
+        Returns
+        -------
+        molecule: oechem.OEGraphMol
+            A protein as OpenEye molecule.
+        """
+        from openeye import oechem
+
+        suffix = path.split(".")[-1]
+        with oechem.oemolistream(path) as ifs:
+            if suffix == "pdb":
+                ifs.SetFlavor(
+                    oechem.OEFormat_PDB,
+                    oechem.OEIFlavor_PDB_Default
+                    | oechem.OEIFlavor_PDB_DATA
+                    | oechem.OEIFlavor_PDB_ALTLOC,
+                )
+            elif suffix == "cif":
+                ifs.SetFlavor(oechem.OEFormat_MMCIF, oechem.OEIFlavor_MMCIF_Default)
+            # TODO: add reasonable defaults for other file formats
+
+            molecule = oechem.OEGraphMol()
+            oechem.OEReadMolecule(ifs, molecule)
+
+        # TODO: returns molecule with 0 atoms if something goes wrong
+        return molecule
+
+    @staticmethod
+    def _write_molecules(molecules: List[oechem.OEGraphMol], path: str):
+        """
+        Save molecules to file.
+        Parameters
+        ----------
+        molecules: list of oechem.OEGraphMol
+            A list of OpenEye molecules for writing.
+        path: str
+            File path for saving molecules.
+        """
+        from openeye import oechem
+
+        with oechem.oemolostream(path) as ofs:
+            for molecule in molecules:
+                oechem.OEWriteMolecule(ofs, molecule)
+        return
+
+    @staticmethod
+    def _has_ligand(molecule: oechem.OEGraphMol) -> bool:
```

Comment:
Should this be extended to include if there is more than one ligand in the system or are we assuming we will only have one ligand present?

Reply:
we could assume that the OpenEyesProteinLigandDockingFeaturizer works with apo proteins or complexes with exactly one ligand in the right place